src/Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests/EventPullerTests.cs

```diff
@@ -0,0 +1,270 @@
+﻿using AutoBogus;
+using Azure.Messaging;
+using FakeItEasy;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Workleap.DomainEventPropagation.EventGridClientAdapter;
+
+namespace Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests;
+
+// Test class cannot be made static
+#pragma warning disable CA1052
+public class EventPullerTests
+#pragma warning restore CA1052
+{
+    private static IEventGridClientAdapter GivenFakeClient(IEventGridClientWrapperFactory clientWrapperFactory, string subName)
+    {
+        var fakeClient = A.Fake<IEventGridClientAdapter>();
+        A.CallTo(() => clientWrapperFactory.CreateClient(subName)).Returns(fakeClient);
+        return fakeClient;
+    }
+
+    private static IServiceScopeFactory GivenScopeFactory(ICloudEventHandler handler)
+    {
+        var scopeFactory = A.Fake<IServiceScopeFactory>();
+        var scope = A.Fake<IServiceScope>();
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddScoped<ICloudEventHandler>(_ => handler);
+        A.CallTo(() => scopeFactory.CreateScope()).Returns(scope);
+        A.CallTo(() => scope.ServiceProvider).Returns(serviceCollection.BuildServiceProvider());
+        return scopeFactory;
+    }
+
+    private static EventPropagationSubscriptionOptions GivenEventPropagationSubscriptionOptions(IOptionsMonitor<EventPropagationSubscriptionOptions> optionMonitor, string clientName)
+    {
+        var option = AutoFaker.Generate<EventPropagationSubscriptionOptions>();
+        A.CallTo(() => optionMonitor.Get(clientName)).Returns(option);
+        return option;
+    }
+
+    private static void RegisterCloudEventsInClient(IEventGridClientAdapter client, EventPropagationSubscriptionOptions option, params EventGridClientAdapter.EventGridClientAdapter.EventBundle[] events)
+    {
+        A.CallTo(() => client.ReceiveCloudEventsAsync(
+            option.TopicName,
+            option.SubscriptionName,
+            A<CancellationToken>._)).Returns(events);
+    }
+
+    private static async Task StartWaitAndStop(IHostedService puller)
+    {
+// We need this to start on the thread pool otherwise it will just block the test
+#pragma warning disable CS4014
+        Task.Run(() => puller.StartAsync(CancellationToken.None));
```

Comment:
T'aurais pas le même behavior et pas de warning en faisant genre
```
var task = puller.StartAsync(CancellationToken.None);
await Task.Delay(50);
await puller.StopAsync(CancellationToken.None);
```

Reply:
In #114, we added an extension method (scoped to tests at the moment) to safely execute a task in "Fire & forget" fashion. Warnings are gone now.